File: java/src/main/java/com/meteroid/api/OAuthApps.java
// this file is @generated
package com.meteroid.api;

import com.meteroid.MeteroidHttpClient;
import com.meteroid.exceptions.ApiException;
import com.meteroid.models.CreateOAuthAppRequest;
import com.meteroid.models.OAuthApp;
import com.meteroid.models.OAuthAppWithSecret;
import com.meteroid.models.OAuthAppsResponse;
import com.meteroid.models.RotatedSecret;

import okhttp3.HttpUrl;

import java.io.IOException;

public class OAuthApps {
    private final MeteroidHttpClient client;

    public OAuthApps(MeteroidHttpClient client) {
        this.client = client;
    }

    /** List all OAuth applications registered for this platform. */
    public OAuthAppsResponse listOauthApps() throws IOException, ApiException {

        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/oauth-apps");
        return this.client.executeRequest("GET", url.build(), null, null, OAuthAppsResponse.class);
    }

    /**
     * Register a new OAuth application. Returns the app with its client secret (only shown once).
     */
    public OAuthAppWithSecret createOauthApp(final CreateOAuthAppRequest createOAuthAppRequest)
            throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/oauth-apps");
        return this.client.executeRequest(
                "POST", url.build(), null, createOAuthAppRequest, OAuthAppWithSecret.class);
    }

    /** Retrieve an OAuth application by ID. */
    public OAuthApp getOauthApp(final String id) throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client.newUrlBuilder().encodedPath(String.format("/api/v1/oauth-apps/%s", id));
        return this.client.executeRequest("GET", url.build(), null, null, OAuthApp.class);
    }

    /** Delete an OAuth application and revoke all associated tokens. */
    public void deleteOauthApp(final String id) throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client.newUrlBuilder().encodedPath(String.format("/api/v1/oauth-apps/%s", id));
        this.client.executeRequest("DELETE", url.build(), null, null, null);
    }

    /** Generate a new client secret for an OAuth app. The old secret is immediately invalidated. */
    public RotatedSecret rotateClientSecret(final String id) throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client
                        .newUrlBuilder()
                        .encodedPath(String.format("/api/v1/oauth-apps/%s/rotate", id));
        return this.client.executeRequest("POST", url.build(), null, null, RotatedSecret.class);
    }
}

File: python/meteroid/api/o_auth_apps.py
# this file is @generated

from ..models import (
    CreateOAuthAppRequest,
    OAuthApp,
    OAuthAppsResponse,
    OAuthAppWithSecret,
    RotatedSecret,
)
from .common import ApiBaseAsync, ApiBaseSync, decode_response


class OAuthAppsAsync(ApiBaseAsync):
    """o auth apps API."""

    async def list_oauth_apps(
        self,
    ) -> OAuthAppsResponse:
        """List all OAuth applications registered for this platform."""
        response = await self._request_asyncio(
            method="get",
            path="/api/v1/oauth-apps",
        )
        return decode_response(response, OAuthAppsResponse)

    async def create_oauth_app(
        self,
        create_o_auth_app_request: CreateOAuthAppRequest,
    ) -> OAuthAppWithSecret:
        """Register a new OAuth application. Returns the app with its client secret
        (only shown once)."""
        response = await self._request_asyncio(
            method="post",
            path="/api/v1/oauth-apps",
            json_body=create_o_auth_app_request.to_dict(),
        )
        return decode_response(response, OAuthAppWithSecret)

    async def get_oauth_app(
        self,
        id: str,
    ) -> OAuthApp:
        """Retrieve an OAuth application by ID."""
        response = await self._request_asyncio(
            method="get",
            path="/api/v1/oauth-apps/{id}",
            path_params={
                "id": id,
            },
        )
        return decode_response(response, OAuthApp)

    async def delete_oauth_app(
        self,
        id: str,
    ) -> None:
        """Delete an OAuth application and revoke all associated tokens."""
        await self._request_asyncio(
            method="delete",
            path="/api/v1/oauth-apps/{id}",
            path_params={
                "id": id,
            },
        )

    async def rotate_client_secret(
        self,
        id: str,
    ) -> RotatedSecret:
        """Generate a new client secret for an OAuth app. The old secret is
        immediately invalidated."""
        response = await self._request_asyncio(
            method="post",
            path="/api/v1/oauth-apps/{id}/rotate",
            path_params={
                "id": id,
            },
        )
        return decode_response(response, RotatedSecret)


class OAuthApps(ApiBaseSync):
    """o auth apps API."""

    def list_oauth_apps(
        self,
    ) -> OAuthAppsResponse:
        """List all OAuth applications registered for this platform."""
        response = self._request_sync(
            method="get",
            path="/api/v1/oauth-apps",
        )
        return decode_response(response, OAuthAppsResponse)

    def create_oauth_app(
        self,
        create_o_auth_app_request: CreateOAuthAppRequest,
    ) -> OAuthAppWithSecret:
        """Register a new OAuth application. Returns the app with its client secret
        (only shown once)."""
        response = self._request_sync(
            method="post",
            path="/api/v1/oauth-apps",
            json_body=create_o_auth_app_request.to_dict(),
        )
        return decode_response(response, OAuthAppWithSecret)

    def get_oauth_app(
        self,
        id: str,
    ) -> OAuthApp:
        """Retrieve an OAuth application by ID."""
        response = self._request_sync(
            method="get",
            path="/api/v1/oauth-apps/{id}",
            path_params={
                "id": id,
            },
        )
        return decode_response(response, OAuthApp)

    def delete_oauth_app(
        self,
        id: str,
    ) -> None:
        """Delete an OAuth application and revoke all associated tokens."""
        self._request_sync(
            method="delete",
            path="/api/v1/oauth-apps/{id}",
            path_params={
                "id": id,
            },
        )

    def rotate_client_secret(
        self,
        id: str,
    ) -> RotatedSecret:
        """Generate a new client secret for an OAuth app. The old secret is
        immediately invalidated."""
        response = self._request_sync(
            method="post",
            path="/api/v1/oauth-apps/{id}/rotate",
            path_params={
                "id": id,
            },
        )
        return decode_response(response, RotatedSecret)
